### PR TITLE
More code improvements

### PR DESCRIPTION
**Pull request by natewind**
Finally found time to finish https://github.com/sourcehold/Sourcehold/pull/31.

### CODE REVIEW

File: src/GameMap.cpp [hunk 1]
```diff
@@ -39,10 +40,11 @@ void GameMap::LoadFromDisk(ghc::filesystem::path path) {
   std::mt19937 rng(dev());
   std::uniform_int_distribution<std::mt19937::result_type> dist(0, 8);
 
-  for (uint32_t i = 0; i < tiles.size(); i++) {
-    int tile = dist(rng);
-    tiles[i] = tileset->GetTile(tile);
-  }
+  auto get_tile = [&]() {
```
Comment (skrax) [inline src/GameMap.cpp:43]: GetRandomTile

File: src/Parsers/Gm1File.cpp [hunk 2]
```diff
@@ -148,8 +148,8 @@ bool Gm1File::LoadFromDisk(ghc::filesystem::path path, bool cached) {
 
   /* Read compressed images into buffer */
   uint32_t buflen = Parser::GetLength() - offData;
-  char *imgdata = new char[buflen];
-  Parser::GetData(imgdata, buflen);
+  auto imgdata = std::vector<char>(buflen);
+  Parser::GetData(imgdata.data(), buflen);
```
Comment (skrax) [inline src/Parsers/Gm1File.cpp:152]: image_buffer

File: src/Parsers/Gm1File.cpp [hunk 3]
```diff
@@ -148,8 +148,8 @@ bool Gm1File::LoadFromDisk(ghc::filesystem::path path, bool cached) {
 
   /* Read compressed images into buffer */
   uint32_t buflen = Parser::GetLength() - offData;
```
Comment (skrax) [inline src/Parsers/Gm1File.cpp:150]: size_of_images

File: src/Parsers/TgxFile.cpp [hunk 4]
```diff
@@ -39,18 +41,17 @@ bool TgxFile::LoadFromDisk(ghc::filesystem::path path) {
   /* Calculate size */
   size_t size = Parser::GetLength() - Parser::Tell();
```
Comment (skrax) [inline src/Parsers/TgxFile.cpp:42]: texture_size

File: src/Parsers/TgxFile.cpp [hunk 5]
```diff
@@ -39,18 +41,17 @@ bool TgxFile::LoadFromDisk(ghc::filesystem::path path) {
   /* Calculate size */
   size_t size = Parser::GetLength() - Parser::Tell();
 
-  char *buf = new char[size];
-  Parser::GetData(buf, size);
+  auto buf = std::vector<char>(size);
+  Parser::GetData(buf.data(), size);
```
Comment (skrax) [inline src/Parsers/TgxFile.cpp:45]: texture_buffer

File: src/Parsers/Gm1File.cpp (hunk 3 above)
Comment (skrax) [reply src/Parsers/Gm1File.cpp:150]: compressed textures ?

**Review by skrax — CHANGES_REQUESTED** [review]
Fine to me, except improve on naming variables and functions correctly.
I created some issues to work on, please take a look or create a new one to discuss.

**Comment by natewind** [comment]
> Fine to me, except improvements on naming variables and functions.
> I created some issues to work on, please take a look or create a new one to discuss.

Fixed the first one, but the rest are unrelated to my changes: the names were already there. Seems like they belong in a separate pool request, to be honest.

**Comment by skrax** [comment]
@natewind You touched the code, so it's up for review.

**Comment by natewind** [comment]
Ok

File: src/Parsers/Gm1File.cpp (hunk 3 above)
Comment (natewind) [reply src/Parsers/Gm1File.cpp:150]: How about `n_textures`? It’s the number of textures, not their size.

Comment (skrax) [reply src/Parsers/Gm1File.cpp]: You are reading raw bytes here.

Comment (skrax) [reply src/Parsers/Gm1File.cpp]: See line 163

Comment (natewind) [reply src/Parsers/Gm1File.cpp]: Ah, ok.

**Review by skrax — APPROVED** [review]
Good changes.

**Comment by natewind** [comment]
@gynt, please merge?

**Pull request merged**